app/src/androidTest/java/com/github/studydistractor/sdp/DistractionActivityTest.kt

```diff
@@ -44,7 +44,7 @@ inline fun <reified A: Activity> ComposeTestRule.launch(
  * See [testing documentation](http://d.android.com/tools/testing).
  */
 @RunWith(AndroidJUnit4::class)
-class ProcrastinationActivityActivityTest {
+class DistractionActivityTest {
```

Comment:
I can't just delete DistractionActivity or the MapActivity won't work anymore

Reply:
Oh, I get it. Here is what i propose :
1. You do your app as a screen, ignoring the fact that map needs it.
2. You create an activity DistractionActivityWrapper in the package "tempActivityWrappers" that will be temporary and that will only contain that code :
```kotlin
@Inject
    lateinit var distractionService : DistractionService
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            DistractionScreen
        }
    }
```

And of course it will lack the top and bottom bar, but that will only be temporary. It's fine because we still get all the functionality, with the sacrifice of a screen being less clean (but that workflow needs refactoring anyway), and at the benifit of you having done really clean work :)

Reply:
I can't do it like that, to know which distraction to display, I pass around a viewModel. I will do in my way but it will be hacky